blocksync/blocksync.go

```diff
@@ -323,13 +317,21 @@ func (bs *blockSyncer) ProcessSyncRequest(ctx context.Context, peer peer.AddrInf
 
 func (bs *blockSyncer) syncStageChecker() {
 	tipHeight := bs.tipHeightHandler()
+	// if tipHeight is equal to targetHeight, it means tried to sync to the tip, but failed.
+	// we need to trigger a sync task
+	if tipHeight == bs.syncStageHeight {
+		bs.syncTask.Trigger()
+	}
```

Comment:
this is just to report the status, let's not do the check and trigger sync here (which is the work of the new TriggerTask)